core/trino-main/src/main/java/io/trino/operator/table/SequenceFunction.java

```diff
@@ -48,93 +45,83 @@
 import static com.google.common.base.Preconditions.checkState;
 import static io.airlift.slice.SizeOf.instanceSize;
 import static io.trino.metadata.GlobalFunctionCatalog.BUILTIN_SCHEMA;
-import static io.trino.operator.table.Sequence.SequenceFunctionSplit.MAX_SPLIT_SIZE;
+import static io.trino.operator.table.SequenceFunction.SequenceFunctionSplit.MAX_SPLIT_SIZE;
 import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
 import static io.trino.spi.function.table.Descriptor.descriptor;
 import static io.trino.spi.function.table.TableFunctionProcessorState.Finished.FINISHED;
 import static io.trino.spi.function.table.TableFunctionProcessorState.Processed.produced;
 import static io.trino.spi.type.BigintType.BIGINT;
 import static java.lang.String.format;
 
-public class Sequence
-        implements Provider<ConnectorTableFunction>
+public class SequenceFunction
+        extends AbstractConnectorTableFunction
 {
     public static final String NAME = "sequence";
 
-    @Override
-    public ConnectorTableFunction get()
+    private static final String START_ARGUMENT_NAME = "START";
+    private static final String STOP_ARGUMENT_NAME = "STOP";
+    private static final String STEP_ARGUMENT_NAME = "STEP";
+
+    public SequenceFunction()
     {
-        return new ClassLoaderSafeConnectorTableFunction(new SequenceFunction(), getClass().getClassLoader());
+        super(
```

Comment:
This is not a new code. It just became a top level class (a rename from Sequence.SequenceFunction to SequenceFunction, no need for Provider).

Nice that github shows this as a rename. Review ignoring whitespace changes to see a small diff.